src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    let newState = { ...currentState };
+
+    if (type === 'addProperties') {
+      for (const key in extraData) {
+        newState[key] = extraData[key];
+      }
+    } else if (type === 'removeProperties') {
+      for (const key of keysToRemove) {
+        delete newState[key];
+      }
+    } else if (type === 'clear') {
+      newState = {};
+    }
```

Comment:
- is better to use switch case statement if instead of multiple if blocks
- move addProperties, removeProperties, clear to the constants as it will make a code cleaner